storage/lmdb/src/resize_callback.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use lmdb::DatabaseResizeInfo;
+
+pub type MapResizeCallbackFn = dyn Fn(DatabaseResizeInfo);
+
+/// A function wrapper that represents the callback when the map within lmdb resizes
+#[derive(Default)]
+pub struct MapResizeCallback {
+    f: Option<Box<MapResizeCallbackFn>>,
```

Comment:
Opposite suggestion to the previous one :laughing:... How about getting rid of the `Option` and just pass in a function that does nothing in the `default` constructor?

Reply:
This all depends on your assumptions... I agree with you on the other one, but this one maps better with the Option because it calls nothing (even an empty function) when it's empty... there's an argument to be made that calling nothing is nothing... but again, all about the assumptions. In this one, I'm mapping 1:1 with what we have in lmdb.

Reply:
Ok, I don't feel very strongly about this.